L1Trigger/L1THGCal/plugins/backend/HGCalSortingTruncationWrapper.cc

```diff
@@ -0,0 +1,134 @@
+  l1thgcfirmware::sortingTruncationAlgoConfig theConfiguration_;
+
+  unsigned kClusterIdMask = 0xFFFFF;
+  unsigned kMulticlusterIdMask = 0xFFFFF;
+  unsigned kClusterIdOffset = 0xFFF;
+  unsigned kMulticlusterIdOffset = 20;
+};
+
+HGCalSortingTruncationWrapper::HGCalSortingTruncationWrapper(const edm::ParameterSet& conf)
+    : HGCalSortingTruncationWrapperBase(conf), theAlgo_(), theConfiguration_(conf.getParameter<unsigned>("maxTCs")) {}
+
+void HGCalSortingTruncationWrapper::convertCMSSWInputs(
+    const l1t::HGCalMulticlusterBxCollection& multiclusters,
+    l1thgcfirmware::HGCalMulticlusterSACollection& multiclusters_SA) const {
+  //std::vector<l1t::HGCalMulticluster> multiclusters;
+
+  multiclusters_SA.reserve(multiclusters.size());
+
+  for (unsigned int imulticluster = 0; imulticluster < multiclusters.size(); ++imulticluster) {
+    bool firstConstituent = true;
+    unsigned int clusterIndex = 0;
+    for (const auto& constituent : multiclusters[imulticluster].constituents()) {
+      if (firstConstituent) {
+        multiclusters_SA.emplace_back(l1thgcfirmware::HGCalMulticluster(
+            l1thgcfirmware::HGCalCluster(constituent.second->centreProj().x(),
+                                         constituent.second->centreProj().y(),
+                                         constituent.second->centreProj().z(),
+                                         triggerTools_.zside(constituent.second->detId()),
+                                         triggerTools_.layerWithOffset(constituent.second->detId()),
+                                         constituent.second->eta(),
+                                         constituent.second->phi(),
+                                         constituent.second->pt(),
+                                         constituent.second->mipPt(),
+                                         imulticluster),
+            1.));
+
+      } else {
+        multiclusters_SA.at(imulticluster)
+            .addConstituent(l1thgcfirmware::HGCalCluster(constituent.second->centreProj().x(),
+                                                         constituent.second->centreProj().y(),
+                                                         constituent.second->centreProj().z(),
+                                                         triggerTools_.zside(constituent.second->detId()),
+                                                         triggerTools_.layerWithOffset(constituent.second->detId()),
+                                                         constituent.second->eta(),
+                                                         constituent.second->phi(),
+                                                         constituent.second->pt(),
+                                                         constituent.second->mipPt(),
+                                                         imulticluster),
+                            1.);
+      }
+      firstConstituent = false;
+      ++clusterIndex;
```

Comment:
This `clusterIndex` doesn't seem to be used

Reply:
Removed